api/services/Ban.js

```diff
@@ -0,0 +1,180 @@
+              reject({error: 'Failed to update /r/' + subreddit + ' AutoModerator config'});
+            } else {
+              resolve('Added /u/' + username + '\'s friend codes to /r/' + subreddit + ' AutoModerator blacklist');
+              console.log('Added /u/' + username + '\'s friend codes to /r/' + subreddit + ' AutoModerator blacklist');
+            }
+          }
+        );
+      }
+    }
+  );
+};
+//Remove the user's TSV threads on /r/SVExchange.
+exports.removeTSVThreads = function(redToken, username, resolve, reject) {
+  Reddit.searchTSVThreads(
+    redToken,
+    username,
+    function (err, response) {
+      if (err) {
+        console.log(err);
+        reject({error: 'Failed to search for user\'s TSV threads'});
+      } else {
+        response.data.children.forEach(function (entry) {
+          Reddit.removePost(
+            redToken,
+            entry.data.id,
+            function (err) {
+              if (err) {
+                console.log(err);
+                reject({error: 'Failed to remove the TSV thread at redd.it/' + entry.data.id});
+              }
+            }
+          );
+        });
+        resolve('Removed /u/' + username + '\'s TSV threads (' + response.data.children.length.toString() + ' total)');
+        console.log('Removed /u/' + username + '\'s TSV threads (' + response.data.children.length.toString() + 'total)');
+      }
+    }
+  );
+};
+//Update the public banlist with the user's information
+exports.updateBanlist = function (redToken, username, banlistEntry, friend_codes, igns, resolve, reject) {
+  Reddit.getWikiPage(
+    redToken,
+    'pokemontrades',
+    'banlist',
+    function (err, current_list) {
+      if (err) {
+        console.log(err);
+        reject({error: 'Failed to retrieve current banlist'});
+        return;
+      }
+      else {
+        var lines = current_list.data.content_md.replace(/\r/g, '').split("\n");
+        var start_index = lines.indexOf('[//]:# (BEGIN BANLIST)') + 3;
+        if (start_index == 2) {
+          console.log('Error: Could not find start marker in public banlist');
+          reject({error: 'Error while parsing public banlist'});
+          return;
+        }
+        var line_to_add = '/u/' + username + ' | ' + friend_codes.join(', ') + ' | ' + banlistEntry + ' | ' + igns;
+        var content = lines.slice(0,start_index).join("\n") + "\n" + line_to_add + "\n" + lines.slice(start_index).join("\n");
+        Reddit.editWikiPage(
+          redToken,
+          'pokemontrades',
+          'banlist',
+          content,
+          '',
+          function (err, response) {
+            if (err) {
+              console.log(err);
+              reject({error: 'Failed to update public banlist'});
+            } else {
+              resolve('Added /u/' + username + ' to public banlist');
+              console.log('Added /u/' + username + ' to public banlist');
+            }
+          }
+        );
+      }
+    }
+  );
+};
```

Comment:
I decided to keep the backend for this even though we removed the button for locally banning/unbanning people. We shouldn't have to use this too often, but it could come in handy if we unban someone after an appeal. (Without this, there's no easy way to remove a user from the local banlist.)